packages/core/src/components/box/boxProps.ts

```diff
@@ -0,0 +1,330 @@
+    | "end"
+    | "center"
+    | "between"
+    | "around"
+    | "evenly"
+    | "normal"
+    | "baseline"
+    | "stretch";
+
+/**
+ * The range of values for `align-items`.
+ *
+ * @see https://developer.mozilla.org/en-US/docs/Web/CSS/align-items
+ */
+export type AlignItems = "start" | "end" | "center" | "baseline" | "stretch";
+
+/**
+ * The range of values for `align-self`.
+ *
+ * @see https://developer.mozilla.org/en-US/docs/Web/CSS/align-self
+ */
+export type AlignSelf = "auto" | "start" | "end" | "center" | "baseline" | "stretch";
+
+/**
+ * The range of values for `display`.
+ *
+ * @see https://developer.mozilla.org/en-US/docs/Web/CSS/display
+ */
+export type Display =
+    | "block"
+    | "inline-block"
+    | "inline"
+    | "flex"
+    | "inline-flex"
+    | "grid"
+    | "inline-grid"
+    | "table"
+    | "inline-table"
+    | "table-cell"
+    | "table-row"
+    | "none";
+
+/**
+ * The range of values for `flex`.
+ *
+ * @see https://developer.mozilla.org/en-US/docs/Web/CSS/flex
+ */
+export type Flex = 1 | "1" | "auto" | "initial" | "none";
+
+/**
+ * The range of values for `flex-direction`.
+ *
+ * @see https://developer.mozilla.org/en-US/docs/Web/CSS/flex-direction
+ */
+export type FlexDirection = "row" | "row-reverse" | "column" | "column-reverse";
+
+/**
+ * The range of values for `flex-wrap`.
+ *
+ * @see https://developer.mozilla.org/en-US/docs/Web/CSS/flex-wrap
+ */
+export type FlexWrap = "nowrap" | "wrap" | "wrap-reverse";
+
+/**
+ * The range of values for `justify-content`.
+ *
+ * @see https://developer.mozilla.org/en-US/docs/Web/CSS/justify-content
+ */
+export type JustifyContent = "start" | "end" | "center" | "between" | "around" | "evenly" | "normal" | "stretch";
+
+/**
+ * The range of values for `justify-items`.
+ *
+ * @see https://developer.mozilla.org/en-US/docs/Web/CSS/justify-items
+ */
+export type JustifyItems = "start" | "end" | "center" | "stretch";
+
+/**
+ * The range of values for `justify-self`.
+ *
+ * @see https://developer.mozilla.org/en-US/docs/Web/CSS/justify-self
+ */
+export type JustifySelf = "auto" | "start" | "end" | "center" | "stretch";
+
+/**
+ * The range of values for `overflow`.
+ *
+ * @see https://developer.mozilla.org/en-US/docs/Web/CSS/overflow
+ */
+export type Overflow = "visible" | "hidden" | "scroll" | "auto";
+
+/**
+ * The range of values for `position`.
+ *
+ * @see https://developer.mozilla.org/en-US/docs/Web/CSS/position
+ */
+export type Position = "static" | "absolute" | "relative" | "fixed" | "sticky";
+
+type BoxOwnProps = Readonly<{
+    /**
+     * Underlying element or component to render. Accepts any intrinsic
+     * element name (e.g. "div", "span") or a React component.
+     *
+     * Defaults to "div" when omitted.
+     */
+    as?: As;
+
+    /**
+     * When true, `<Box>` will **not** render an extra wrapper. Instead, it will
+     * clone & enhance its single child element, merging class names and props.
+     */
+    asChild?: boolean;
```

Comment:
Wondering if we can combine these by having `"child"` be an option for `As` so that we don't have a weird situation where a user has both `asChild` and `as` and we need to determine what that means.